packages/sdk-client-v3/src/client/client.ts

```diff
@@ -162,11 +162,9 @@ export default function createClient(middlewares: ClientOptions): Client {
       } as MiddlewareResponse
 
       if (res.error) {
-        res.reject(res.error)
```

Comment:
Don't you think it will be better to return the error here instead of throwing it, so that the ErrorMiddleware can properly construct the error based on this returned value?

Reply:
https://github.com/commercetools/commercetools-sdk-typescript/pull/617/commits/82ef8ac299dea3ed4e3f7c7e5184e34213524472